azurerm/internal/services/eventgrid/eventgrid_event_subscription_resource_test.go

```diff
@@ -743,3 +782,63 @@ resource "azurerm_eventgrid_event_subscription" "test" {
 }
 `, data.RandomInteger, data.Locations.Primary, data.RandomString, data.RandomInteger, data.RandomInteger)
 }
+
+func (EventGridEventSubscriptionResource) identity(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-eg-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "acctestacc%[3]s"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+
+  tags = {
+    environment = "staging"
+  }
+}
+
+resource "azurerm_storage_queue" "test" {
+  name                 = "mysamplequeue-%[1]d"
+  storage_account_name = azurerm_storage_account.test.name
+}
+
+resource "azurerm_storage_container" "test" {
+  name                  = "dead_letter_destination"
```

Comment:
This test is failing because  we can't have a name in this format
```

testing.go:620: Step 1/6 error: Error running pre-apply refresh: exit status 1
Error: only lowercase alphanumeric characters and hyphens allowed in "name": "dead_letter_destination"
with azurerm_storage_container.test,
on terraform_plugin_test.tf line 28, in resource "azurerm_storage_container" "test":
28: resource "azurerm_storage_container" "test" {
--- FAIL: TestAccEventGridEventSubscription_identity (15.53s)
```